src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -234,10 +236,13 @@ void GcsActorManager::HandleRegisterActor(const rpc::RegisterActorRequest &reque
   RAY_LOG(INFO) << "Registering actor, job id = " << actor_id.JobId()
                 << ", actor id = " << actor_id;
   Status status =
-      RegisterActor(request, [reply, send_reply_callback,
+      RegisterActor(request, [this, reply, send_reply_callback,
                               actor_id](const std::shared_ptr<gcs::GcsActor> &actor) {
         RAY_LOG(INFO) << "Registered actor, job id = " << actor_id.JobId()
                       << ", actor id = " << actor_id;
+        if (actor->IsDetached()) {
+          function_manager_.AddJobReference(actor_id.JobId());
```

Comment:
How about we always add the reference?

Reply:
Could you share some ideas about why change it in that way?

Reply:
This is just to avoid special cases in the code. If we always add it, it's more likely we catch bugs. But if there is a performance penalty, we can justify having a special case.

Reply:
I don't think there is a big performance penalty here. The only thing is a hash map lookup I think.

I don't have a preference here, but this defines what does a job ended means:

1. driver exits + all actors exits
1. driver exits + all detached actors exit.

Reply:
We kill actors on exit right? So it seems about the same, unless this would interfere.

Reply:
from the code, they are probably the same for now, unless we changed this in the future (we kill only when no place is referencing the actor).

yeah, as I mentioned, I don't have a preference here. either works for me.